src/base/bittorrent/bencoderesumedatastorage.cpp

```diff
@@ -134,42 +134,84 @@ QVector<BitTorrent::TorrentID> BitTorrent::BencodeResumeDataStorage::registeredT
     return m_registeredTorrents;
 }
 
-std::optional<BitTorrent::LoadTorrentParams> BitTorrent::BencodeResumeDataStorage::load(const TorrentID &id) const
+BitTorrent::LoadResumeDataResult BitTorrent::BencodeResumeDataStorage::load(const TorrentID &id) const
 {
     const QString idString = id.toString();
     const Path fastresumePath = m_resumeDataPath / Path(idString + u".fastresume");
     const Path torrentFilePath = m_resumeDataPath / Path(idString + u".torrent");
 
     QFile resumeDataFile {fastresumePath.data()};
     if (!resumeDataFile.open(QIODevice::ReadOnly))
-    {
-        LogMsg(tr("Cannot read file %1: %2").arg(fastresumePath.toString(), resumeDataFile.errorString()), Log::WARNING);
-        return std::nullopt;
-    }
+        return nonstd::make_unexpected(tr("Cannot read file %1: %2").arg(fastresumePath.toString(), resumeDataFile.errorString()));
 
     QFile metadataFile {torrentFilePath.data()};
     if (metadataFile.exists() && !metadataFile.open(QIODevice::ReadOnly))
-    {
-        LogMsg(tr("Cannot read file %1: %2").arg(torrentFilePath.toString(), metadataFile.errorString()), Log::WARNING);
-        return std::nullopt;
-    }
+        return nonstd::make_unexpected(tr("Cannot read file %1: %2").arg(torrentFilePath.toString(), metadataFile.errorString()));
 
     const QByteArray data = resumeDataFile.readAll();
     const QByteArray metadata = (metadataFile.isOpen() ? metadataFile.readAll() : "");
 
     return loadTorrentResumeData(data, metadata);
 }
 
-std::optional<BitTorrent::LoadTorrentParams> BitTorrent::BencodeResumeDataStorage::loadTorrentResumeData(
-        const QByteArray &data, const QByteArray &metadata) const
+void BitTorrent::BencodeResumeDataStorage::doLoadAll() const
+{
+    qDebug() << "Loading torrents count: " << m_registeredTorrents.size();
+
+    emit const_cast<BencodeResumeDataStorage *>(this)->loadStarted(m_registeredTorrents);
+
+    for (const TorrentID &torrentID : asConst(m_registeredTorrents))
+    {
+        nonstd::expected<LoadTorrentParams, QString> loadResumeDataResult = load(torrentID);
```

Comment:
If I read correctly:
```suggestion
        const LoadResumeDataResult loadResumeDataResult = load(torrentID);
```
Or just inline the call into the next line.

Reply:
> Or just inline the call into the next line.

Done.